sqlmesh/core/scheduler.py

```diff
@@ -55,9 +55,46 @@
 
 logger = logging.getLogger(__name__)
 SnapshotToIntervals = t.Dict[Snapshot, Intervals]
-# we store snapshot name instead of snapshots/snapshotids because pydantic
-# is extremely slow to hash. snapshot names should be unique within a dag run
-SchedulingUnit = t.Tuple[str, t.Tuple[Interval, int]]
+
+
+class BaseNode:
+    snapshot_name: str
+
+    def __lt__(self, other: BaseNode) -> bool:
+        return (self.__class__.__name__, self.snapshot_name) < (
+            other.__class__.__name__,
+            other.snapshot_name,
+        )
+
+
+@dataclass(frozen=True)
+class EvaluateNode(BaseNode):
+    snapshot_name: str
```

Comment:
This is already defined in `BaseNode`. Same applies for `CreateNode` and `Dummy Node`.